src/project/internal/notationproject.cpp

```diff
@@ -566,9 +566,10 @@ mu::Ret NotationProject::doSave(const io::path_t& path, engraving::MscIoMode ioM
 
     // Step 1: check writable
     {
-        if (fileSystem()->exists(savePath) && !fileSystem()->isWritable(savePath)) {
+        if ((fileSystem()->exists(savePath) && !fileSystem()->isWritable(savePath))
+            || (fileSystem()->exists(targetContainerPath) && !fileSystem()->isWritable(targetContainerPath))) {
             LOGE() << "failed save, not writable path: " << savePath;
-            return make_ret(notation::Err::UnknownError);
+            return make_ret(notation::Err::NotWritable);
```

Comment:
That might work, but these really ain't notation errors, but rather I/O errors

Reply:
I tried to be coherent and use the same namespace (notation) as the file where this verification is made, notationprokect.cpp, uses. Do you think the error should be handled elsewhere?

Reply:
IMHO yes, as these ain't notation errors, but file I/O errors

Reply:
But see https://github.com/musescore/MuseScore/pull/22054#discussion_r1538007993)

Reply:
Thanks again for your valuable contributions. I pushed a new commit changing the error codes. I agree with you, those are not notation, but rather I/O errors. I'll leave it to the team to check if the messages presented to the user should be updated (make them more descriptive), but they are better than "Unknown error" in my opinion.